### PR TITLE
Preemptible support in Open Targets Platform

### PR DESCRIPTION
**Pull request by mbdebian**
This pull requests extends Open Targets Platform provisioner for supporting preemptible VMs.
It also includes a permanent development deployment context with preemptible flags active.
Context for this pull request: opentargets/platform#1980

### CODE REVIEW

File: modules/elasticsearch/main.tf [hunk 1]
```diff
@@ -59,8 +60,10 @@ resource "google_compute_instance_template" "elastic_search_template" {
   can_ip_forward = false
 
   scheduling {
-    automatic_restart = true
-    on_host_maintenance = "MIGRATE"
+    automatic_restart = !var.vm_flag_preemptible
+    on_host_maintenance = var.vm_flag_preemptible ? "TERMINATE" : "MIGRATION"
```
Comment (JarrodBaker) [inline modules/elasticsearch/main.tf:64]: Should 'MIGRATION' be 'MIGRATE'? That is what is was before.

File: modules/webapp/webservers-compute.tf [hunk 2]
```diff
@@ -56,8 +57,10 @@ resource "google_compute_instance_template" "webserver_template" {
   can_ip_forward = false
 
   scheduling {
-    automatic_restart = true
-    on_host_maintenance = "MIGRATE"
+    automatic_restart = !var.vm_flag_preemptible
+    on_host_maintenance = var.vm_flag_preemptible ? "TERMINATE" : "MIGRATION"
```
Comment (JarrodBaker) [inline modules/webapp/webservers-compute.tf:61]: Same as above re 'MIGRATE' vs 'MIGRATION'

Comment (mbdebian) [reply modules/webapp/webservers-compute.tf:61]: Actually it should, damn autocorrect. But the mystery remains in the fact that I tried both and they worked. I’ll change it anyway. Thanks @JarrodBaker !

Comment (mbdebian) [reply modules/webapp/webservers-compute.tf:61]: Done, I pushed to main straight away, I'm not sure why GCP was not complaining about it.

**Pull request merged**